testsuite/gnat2goto/tests/long_integer/test_long_integer.adb

```diff
@@ -0,0 +1,6 @@
+with Ada.Text_IO;
+procedure Test_Long_Integer is
+  x: Long_Long_Integer := 1152921504606846976;
```

Comment:
Check: does cbmc require widths to be multiples of a byte? Does Ada?

Reply:
W.r.t Ada: No, consider

```
with Ada.Text_IO;
procedure Size_Test is

  package IO renames Ada.Text_IO;

  type My_Bit is (On, Off)
    with Size => 1;

  type My_Bits is Array(Integer range <>) of My_Bit
    with Pack;
  
  Bit : My_Bit := Off;

  type My_Rec is record
    Bits : My_Bits (0..7);
  end record;
begin
  IO.Put_Line ("Bit: " & My_Bit'Image (Bit));
  Bit := On;
  IO.Put_Line ("Bit: " & My_Bit'Image (Bit));
  IO.Put_Line ("Bit size: " & Integer'Image (My_Bit'Size));
  IO.Put_Line ("Struct size: " & Integer'Image (My_Rec'Size));
end Size_Test;
```

Output will be

> Bit: OFF
Bit: ON
Bit size:  1
Struct size:  8

As to cbmc, I don't think so? We already support bitfields which require odd sizes, and `bitvector_typet` - which is the base of most primitives - doesn't seem to place any restrictions on the width passed (which is in bits).